src/python/pants/engine/internals/graph.py

```diff
@@ -585,26 +609,6 @@ def compatible_with_sources_field(valid_type: Type[Sources]) -> bool:
 # -----------------------------------------------------------------------------------------------
 
 
-class AmbiguousDependencyInferenceException(Exception):
```

Comment:
Hm.. breaking out feels like overkill to me. If dependency inference was already more widely used (or if we were encouraging people to write dependency inference plugins... I don't think we are), that might be another thing. But this also _loosens_ a restriction rather than tightening it.

Reply:
Fair enough.